### DISCUSSION TITLE
How to build a dynamic test？

### DISCUSSION THREAD

**Discussion by ghost** [jspsych/jsPsych#3005] [Q&A] [answered]
I would like to create a **dynamic test** that contains multiple blocks, each consisting of several Likert items. If the subject gives the same response to two or more items in a block, these items will be presented again in a forced-choice format. Here is an example:
![image](https://user-images.githubusercontent.com/48197584/231723832-15975aae-922c-47be-9101-d3eeb0639919.png)
Are there any plugins or extensions available? Many thanks.

**Comment by ChristopheBossens** [comment, marked as answer]
I would use the survey plugin and create a likert-table to start with. 

My approach would then be to create an `on_finish `function in which you extract the response to the current block extract the questions to which the same answer has been given. I would then create an additional survey block and use a [conditional timeline](https://www.jspsych.org/7.3/overview/timeline/#conditional-timelines) to decide if that block needs to be shown or not. Finally, you should be able to set the pages parameter with a function to create [dynamic parameters](https://www.jspsych.org/7.3/overview/dynamic-parameters/) so that you can fill in the questions based on the results of what you calculate in the `on_finish `function.

I was wondering how you would deal with a situation where you have more than three items, and the participant makes multiple same responses but with different response alternatives (e.g., 2 times strongly disagree and 2 times agree)?

**Reply by ghost** [reply]
First of all, thank you very much for your answer. I will try according to the method you provided.
You raised a very good question. The purpose of this dynamic test is that the items within the same block will get their unique ranking (agreement or preference level). Therefore, when there are multiple groups of the same options, they will all be presented in a forced-choice format until all items have different rankings.
For example, if participants chose strongly agree on two items and strongly disagree on two items, then these two groups of items need to be presented again in a forced-choice manner respectively (I was wondering if this approach would cause trouble in programming? ).

**Reply by ChristopheBossens** [reply]
I think it makes the programming a little bit more complicated but not impossible. As far as I know it is not possible in jsPsych to dynamically insert new trials. You can however use a conditional time line and decide during your experiment if you want to run these trials (in combination with dynamic parameters). Using this mechanism, I think you can prepare as many additional trials as there are groups that can be formed, and decide how many of those you want to run based on the actual response.

**Reply by ghost** [reply]
Thank you very much for your reply!
